crates/types/src/spline_motion_interpolator.rs

```diff
@@ -0,0 +1,107 @@
+use splines::{Interpolation, Key, Spline};
+use thiserror::Error;
+
+use crate::{Joints, MotionFile};
+use std::time::Duration;
+
+pub struct SplineMotionFileInterpolator {
```

Comment:
Could you explain how the types relate together? It looks like you added another type and kept the existing ones. If I understood it correctly, the new spline interpolation only changes the implementation but not the interface. Would it be possible to use the new spline interpolation everywhere and therefore be able to remove the existing types? Deleted code is debugged code. 😉

Reply:
The other motion interpolator is removed now